src/molecules/Dropdown/index.tsx

```diff
@@ -82,7 +94,11 @@ const Dropdown: React.SFC<DropdownProps> = ({
             keepInViewPort={keepInViewPort}
           >
             {flyoutContainer ? (
-              <StyledMenu width={width} data-qaid={`${qaId}-flyout`}>
+              <StyledMenu
+                width={width}
+                ref={menuRef}
```

Comment:
do we also need to add the menuRef in the case where `flyoutContainer == false`? I get kind of unexpected behavior in your storybook story where `flyoutContainer={false}` when `keepOpenOnClick` is `true`, because it's passing `menuRef` into the `useOutsideClick` but not rendering any element with `ref={menuRef}`. Might be able to just do:

```
<div ref={menuRef}>
  {children}
</div>
```